src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,319 @@
+package duke.util;
+
+import duke.Duke;
+
+import duke.exception.InvalidCommandException;
+import duke.exception.MissingInputException;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Event;
+import duke.task.Deadline;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * The Parser class is responsible for parsing user input and executing corresponding commands.
+ * It handles various user commands such as adding tasks, listing tasks, marking tasks as done, un-marking tasks and
+ * deleting tasks.
+ *
+ * @author Win Sheng
+ * @since 3 September 2023
+ */
+public class Parser {
+
+    /**
+     * Parses the user's input and executes the corresponding commands.
+     *
+     * @param userInput The user's input.
+     * @param taskList The list of tasks.
+     * @param ui The Ui for displaying messages.
+     * @param storage The Storage for saving and loading tasks.
+     */
+    public void parseUserInput(String userInput, TaskList taskList, Ui ui, Storage storage) {
+
+        try {
+            String[] individualWords = userInput.split(" ");
+
+            if (individualWords.length > 0) { // To handle whitespace input
+                String firstWord = individualWords[0];
+                String lowerCapsFirstWord = firstWord.toLowerCase();
+
+                try {
+                    switch (lowerCapsFirstWord) {
+                        case "bye":
+                            ui.showGoodbye();
+                            Duke.isDone = true;
+                            break;
+                        case "list":
+                            listTasks(taskList, ui);
+                            break;
+                        case "mark":
+                            markTask(userInput, taskList, ui);
+                            break;
+                        case "unmark":
+                            unmarkTask(userInput, taskList, ui);
+                            break;
+                        case "delete":
+                            deleteTask(userInput, taskList, ui);
+                            break;
+                        case "todo":
+                            addTodoTask(userInput, taskList, ui);
+                            break;
+                        case "deadline":
+                            addDeadlineTask(userInput, taskList, ui);
+                            break;
+                        case "event":
+                            addEventTask(userInput, taskList, ui);
+                            break;
+                        case "find":
+                            findByKeyword(userInput, taskList, ui);
```

Comment:
I think according to the coding standard, case statement belongs on the same line with switch statement?
I think this is what it should look like!
```suggestion
                    switch (lowerCapsFirstWord) {
                    case "bye":
                        ui.showGoodbye();
                        Duke.isDone = true;
                        break;
                    case "list":
                        listTasks(taskList, ui);
                        break;
                    case "mark":
                        markTask(userInput, taskList, ui);
                        break;
                    case "unmark":
                        unmarkTask(userInput, taskList, ui);
                        break;
                    case "delete":
                        deleteTask(userInput, taskList, ui);
                        break;
                    case "todo":
                        addTodoTask(userInput, taskList, ui);
                        break;
                    case "deadline":
                        addDeadlineTask(userInput, taskList, ui);
                        break;
                    case "event":
                        addEventTask(userInput, taskList, ui);
                        break;
                    case "find":
                        findByKeyword(userInput, taskList, ui);
```